components/brave_rewards/browser/rewards_service_observer.h

```diff
@@ -51,6 +53,10 @@ class RewardsServiceObserver : public base::CheckedObserver {
       const ledger::type::RewardsType type,
       const ledger::type::ContributionProcessor processor) {}
 
+  virtual void OnProcessForEnableRewardsStarted() {}
```

Comment:
what is this method supposed to be? Why do we care about something so specific in the startup process for this?

Reply:
This event tells `BraveAdsDriverDelegateImpl` that `Rewards enable` button was clicked and we can resolve promise with `true`.
This is needed because pref enabling takes some time after `Rewards enable` button click. And during this time user can close panel by outside click.

Reply:
I will rename this to OnRewardsEnabled()